third_party/move/tools/move-spec-test/src/report.rs

```diff
@@ -0,0 +1,202 @@
+use serde::Serialize;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use tabled::{builder::Builder, settings::Style};
+
+/// This struct represents a report of the specification testing.
+/// It contains the list of entries, where each entry is a file and the number of mutants tested
+/// and killed in that file (in form of a `ReportEntry` structure).
+#[derive(Debug, Serialize)]
+pub struct Report {
+    /// The list of entries in the report.
+    files: HashMap<PathBuf, MutantStats>,
+}
+
+impl Report {
+    /// Creates a new report.
+    pub fn new() -> Self {
+        Self {
+            files: HashMap::new(),
+        }
+    }
+
+    /// Adds an entry to the report.
+    pub fn add_entry(&mut self, path: PathBuf, entry: MutantStats) {
+        self.files.insert(path, entry);
+    }
+
+    /// Returns `true` if the report contains an entry for the given path.
+    pub fn contains(&self, path: &PathBuf) -> bool {
+        self.files.contains_key(path)
+    }
+
+    /// Increments the number of mutants tested for the given path.
+    /// If the path is not in the report, it adds it with the number of mutants tested set to 1.
+    pub fn increment_mutants_tested(&mut self, path: &PathBuf) {
+        if self.contains(path) {
+            let entry = self.files.get_mut(path).unwrap();
+            entry.tested += 1;
+        } else {
+            self.add_entry(
+                path.clone(),
+                MutantStats {
+                    tested: 1,
+                    killed: 0,
+                },
+            );
+        }
+    }
+
+    /// Increments the number of mutants killed for the given path.
+    /// If the path is not in the report, it adds it with the number of mutants tested and killed set to 1.
+    pub fn increment_mutants_killed(&mut self, path: &PathBuf) {
+        if self.contains(path) {
+            let entry = self.files.get_mut(path).unwrap();
+            entry.killed += 1;
+        } else {
+            self.add_entry(
+                path.clone(),
+                MutantStats {
+                    tested: 1,
+                    killed: 1,
+                },
+            );
+        }
```

Comment:
Something like this should make it simpler (not sure if this exact code will work, but it could be adapted):
```suggestion
    let mut entry = self.files.entry(path).or_insert(MutantStats::default());
    entry.tested += 1;
    entry.killed += 1;
```

Reply:
Hmm, yeah - this can be used for the other function... since I guess you don't want to increment tested whenever killed is incremented, no?
Feel free to resolve this one

Reply:
I've used it for both functions.
I've added additional comment that incrementing killed count does not set tested value to anything other than default.